packages/sourcecred/src/core/storage/github.js

```diff
@@ -0,0 +1,187 @@
+// @flow
+
+import type {DataStorage, WritableDataStorage} from "./index";
+import {stringToRepoId} from "../../plugins/github/repoId";
+import type {GithubToken} from "../../plugins/github/token";
+import type {RepoId, RepoIdString} from "../../plugins/github/repoId";
+import fetch from "cross-fetch";
+import {decode as base64Decode} from "base-64";
+import {decode, encode} from "./textEncoding";
+import {toBase64} from "@aws-sdk/util-base64-browser";
+
+const GET_LEDGER_QUERY = `
+query getLedger($owner: String!, $repo: String!, $expression:String!) {
+  repository(owner: $owner, name: $repo) {
+    object(expression: $expression) {
+      ... on Blob {
+        oid
+      }
+    }
+  }
+}
+`;
+
+type CreateBlobRes = {
+  sha: string,
+  url: string,
+};
+
+type Opts = {
+  apiToken: GithubToken,
+  repo: RepoIdString,
+  branch: string,
+};
+
+export class GithubStorage implements DataStorage {
+  static ENDPOINT: string = "https://api.github.com";
+
+  constructor(opts: Opts) {
+    this._token = opts.apiToken;
+    this._repoId = stringToRepoId(opts.repo);
+    this._branch = opts.branch;
+  }
+
+  _token: GithubToken;
+  _repoId: RepoId;
+  _branch: string;
+
+  /**
+   * data/ledger.json
+   * @param path data/ledger.json
+   * @returns {Promise<Ledger>}
+   */
+  async get(path: string): Promise<Uint8Array> {
+    const options = {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${this._token}`,
+      },
+      body: JSON.stringify({
+        query: GET_LEDGER_QUERY,
+        variables: {
+          owner: this._repoId.owner,
+          repo: this._repoId.name,
+          expression: `${this._branch}:${path}`,
+        },
+      }),
+    };
+
+    const result = await fetch(`${GithubStorage.ENDPOINT}/graphql`, options);
+
+    const {data} = await result.json();
+    const blobSha = data.repository.object.oid;
+
+    const ledgerBlobRes = await fetch(
+      `${this._getRepoEndpoint()}/git/blobs/${blobSha}`,
+      {
+        method: "GET",
+        headers: {
+          Authorization: `Bearer ${this._token}`,
+        },
+      }
+    );
+
+    const ledgerDataBase64 = await ledgerBlobRes.json();
+    const rawLedger = base64Decode(ledgerDataBase64.content);
+    return encode(rawLedger);
+  }
+
+  _getRepoEndpoint(): string {
+    const {owner, name} = this._repoId;
+    return `${GithubStorage.ENDPOINT}/repos/${owner}/${name}`;
+  }
+}
+
+export class WritableGithubStorage extends GithubStorage implements  WritableDataStorage {
+  async set(path: string, ledger: Uint8Array, message?: string): Promise<void> {
```

Comment:
The code is already generalized, so as discussed, let's generalized the payload param (and its handlers) from `ledger` to `content`.